test/e2e/test_system.py

```diff
@@ -4,603 +4,579 @@
+        self.get_last_system_breadcrumbs()
+        self.check_get_breadcrumbs_success(
+            expected_trail_length=BREADCRUMBS_TRAIL_MAX_LENGTH - 1, expected_full_trail=True
+        )
 
-    system = response.json()
+    def test_get_breadcrumbs_when_trail_length_maximum(self):
+        """Test getting a System's breadcrumbs when the full system trail should be equal to the maximum trail
+        length"""
 
-    assert system == SYSTEM_POST_REQUIRED_ONLY_EXPECTED
+        self.post_nested_systems(BREADCRUMBS_TRAIL_MAX_LENGTH)
+        self.get_last_system_breadcrumbs()
+        self.check_get_breadcrumbs_success(expected_trail_length=BREADCRUMBS_TRAIL_MAX_LENGTH, expected_full_trail=True)
 
+    def test_get_breadcrumbs_when_trail_length_greater_maximum(self):
+        """Test getting a System's breadcrumbs when the full system trail exceeds the maximum trail length"""
 
-def test_create_system_with_valid_parent_id(test_client):
-    """
-    Test creating a System with a valid parent ID
-    """
-    # Parent
-    response = test_client.post("/v1/systems", json=SYSTEM_POST_A)
-    parent_system = response.json()
+        self.post_nested_systems(BREADCRUMBS_TRAIL_MAX_LENGTH + 1)
+        self.get_last_system_breadcrumbs()
+        self.check_get_breadcrumbs_success(
+            expected_trail_length=BREADCRUMBS_TRAIL_MAX_LENGTH, expected_full_trail=False
+        )
 
-    # Child
-    response = test_client.post("/v1/systems", json={**SYSTEM_POST_B, "parent_id": parent_system["id"]})
+    def test_get_breadcrumbs_with_non_existent_id(self):
+        """Test getting a System's breadcrumbs when given a non-existent system id"""
 
-    assert response.status_code == 201
-    system = response.json()
-    assert system == {**SYSTEM_POST_B_EXPECTED, "parent_id": parent_system["id"]}
+        self.get_system_breadcrumbs(str(ObjectId()))
+        self.check_get_breadcrumbs_failed_with_message(404, "System not found")
 
+    def test_get_breadcrumbs_with_invalid_id(self):
+        """Test getting a System's breadcrumbs when given an invalid system id"""
 
-def test_create_system_with_duplicate_name_within_parent(test_client):
-    """
-    Test creating a System with a duplicate name within the parent System
-    """
-    # Parent
-    response = test_client.post("/v1/systems", json=SYSTEM_POST_A)
-    parent_system = response.json()
+        self.get_system_breadcrumbs("invalid_id")
+        self.check_get_breadcrumbs_failed_with_message(404, "System not found")
 
-    # Child - post twice as will have the same name
-    test_client.post("/v1/systems", json={**SYSTEM_POST_B, "parent_id": parent_system["id"]})
-    response = test_client.post("/v1/systems", json={**SYSTEM_POST_B, "parent_id": parent_system["id"]})
 
-    assert response.status_code == 409
-    assert response.json()["detail"] == "A System with the same name already exists within the same parent System"
+class ListDSL(GetBreadcrumbsDSL):
```

Comment:
I think its also worth point out I struggled to come up with a consistent naming system throughout these tests, right now its using `_system` or `_systems` in the method names to distinguish between these and later ones for items where we may be able to reuse logic but its not clear to me until we get that far so could be a case of renaming these then. 

Same for the class and method names - should they be the request type or not? Here I have used the request type for methods, but Get and List inside the class names simply because I didn't see the worth in putting System/Systems after each, but again it may be obvious later which is best.

Reply:
Like you say, it won't be clear until we get that far.